modules/nextflow/src/test/groovy/nextflow/cli/CmdSecretTest.groovy

```diff
@@ -54,10 +55,21 @@ class CmdSecretTest extends Specification {
         field.accessible = true
         def map = (Map<String, String>) field.get(System.getenv())
         map.put('NXF_SECRETS_FILE', secretFile.toString())
+        map.put('NXF_ENABLE_SECRETS', 'true')
```

Comment:
Too hacking into System env inner implementation. I've added SysEnv wrapper to better handler this see [here](https://github.com/nextflow-io/nextflow/blob/bdfa0f6d3bf31ec8afebb1cd041e867ac27a05e3/modules/nf-commons/src/main/nextflow/SysEnv.groovy#L29-L29) and [here](https://github.com/nextflow-io/nextflow/blob/bdfa0f6d3bf31ec8afebb1cd041e867ac27a05e3/modules/nextflow/src/main/groovy/nextflow/secret/LocalSecretsProvider.groovy#L50-L50)


But not understanding why some tests are failing

Reply:
Probably because it's also required in `SecretsLoader`

```
 static boolean isEnabled() {
        System.getenv('NXF_ENABLE_SECRETS')=='true'
    }
```

Reply:
(by other hand, the hack to change `env` values it's "pure gold" , isn't it?)

Reply:
Kind of :)

Reply:
I think the error comes because we are using Memoized in `load` method and we need to refresh the cache between tests 

it seems with this commit is working now

https://github.com/nextflow-io/nextflow/commit/d4537712d8e83b2b7c5e75888723b138f0ddb9a7